dGame/dInventory/ItemSetPassiveAbility.cpp

```diff
@@ -293,8 +293,8 @@ void ItemSetPassiveAbility::OnEnemySmshed() {
 		break;
 	}
 	case ItemSetPassiveAbilityID::ShinobiRank3: {
-		if (equippedCount < 4) return;
-		destroyableComponent->Imagine(3);
+		if (equippedCount < 4 || !target) return;
+		skillComponent->CalculateBehavior(695, 11399, target->GetObjectID());
```

Comment:
Unfortunate hardcoding. Is there some system behind this which would explain why this is necessary, are all set skills behaviors?

Reply:
There is, but to keep this PR small and just as a proof of concept that the behavior itself works, I only chose to replace this one specifically with the behavior like this.  The actual implementation will be far cleaner

Reply:
I can go into more detail if you want for how this is supposed to be implemented, whether you want all of these to be fixed at once, leave this alone for now and just implement the node, or implement the node and this one check.  any I am fine with